test/functional/client_side_encryption_corpus_tests.js

```diff
@@ -0,0 +1,381 @@
+      expect(actual).to.equal(expected);
+      return;
+    }
+
+    const expectedValue = expected.value;
+    const actualValue = actual.value;
+    const expectedJSON = toJSONComp(expectedValue);
+    const actualJSON = toJSONComp(actualValue);
+
+    switch (expected.algo) {
+      case 'det': {
+        expect(actualJSON).to.deep.equal(expectedJSON);
+        break;
+      }
+      case 'rand': {
+        if (expected.allowed === true) {
+          expect(actualJSON).to.not.deep.equal(expectedJSON);
+        }
+        break;
+      }
+      default: {
+        throw new Error('how did you get here?');
+      }
+    }
+
+    if (expected.allowed === true) {
+      return Promise.all([
+        clientEncryption.decrypt(expectedValue),
+        clientEncryption.decrypt(actualValue)
+      ]).then(results => {
+        const decryptedExpectedValue = results[0];
+        const decryptedActualValue = results[1];
+
+        const decryptedExpectedJSON = toJSONComp(decryptedExpectedValue);
+        const decryptedActualJSON = toJSONComp(decryptedActualValue);
+
+        expect(decryptedActualJSON).to.deep.equal(decryptedExpectedJSON);
+      });
+    } else if (expected.allowed === false) {
+      expect(actualJSON).to.deep.equal(expectedJSON);
+    } else {
+      throw new Error('how did you get here?');
+    }
+  }
+
+  before(function() {
+    // 1. Create a MongoClient without encryption enabled (referred to as ``client``).
+    client = this.configuration.newClient({
+      useNewUrlParser: true,
+      useUnifiedTopology: true
+    });
+
+    return Promise.resolve()
+      .then(() => client.connect())
+      .then(() => {
+        // 3. Using ``client``, drop the collection ``admin.datakeys``. Insert the documents `corpus/corpus-key-local.json <../corpus/corpus-key-local.json>`_ and `corpus/corpus-key-aws.json <../corpus/corpus-key-aws.json>`_.
+        const keyDb = client.db(keyVaultDbName);
+        return Promise.resolve()
+          .then(() => keyDb.dropCollection(keyVaultCollName))
+          .catch(() => {})
+          .then(() => keyDb.collection(keyVaultCollName))
+          .then(keyColl => keyColl.insertMany([corpusKeyLocal, corpusKeyAws]));
+      });
+  });
+
+  after(function() {
+    if (client) {
+      return client.close(true);
+    }
+  });
+
+  function defineCorpusTests(corpus, corpusEncryptedExpected, useClientSideSchema) {
+    let clientEncrypted, clientEncryption;
+    beforeEach(function() {
+      return Promise.resolve()
+        .then(() => {
+          // 2. Using ``client``, drop and create the collection ``db.coll`` configured with the included JSON schema `corpus/corpus-schema.json <../corpus/corpus-schema.json>`_.
+          const dataDb = client.db(dataDbName);
+          return Promise.resolve()
+            .then(() => dataDb.dropCollection(dataCollName))
+            .catch(() => {})
+            .then(() => {
+              return dataDb.createCollection(dataCollName, {
```

Comment:
nit: no need for braces here

Reply:
yeah, there are just a few instances of this. This would just be `then(() => dataDb.createCollection...))`